### ISSUE TITLE
Hardcoded Ip Check false positive

### ISSUE THREAD

**Issue by sgallou** [SonarOpenCommunity/sonar-cxx#1135]
Hi
This rule asks me to make my IP configurable in a line of my code exactly for the description filed of the configurable option :

`Poco::Util::Option("webServerIp", "i", "Use a specific web server ip address. Use 0.0.0.0 to accepts connections via all interfaces.")`

I think the case where the string containing not only an IP address should be  compliant. I saw that in the files HardcodedIpCheckTest.java and HardcodedIpCheck.cc, the case is not tested.
I suggest to add in HardcodedIpCheck.cc :

```
    const string  noOnlyAnIp = "Some text before 0.0.0.0"; // Compliant
    const string  noOnlyAnIp2 = "0.0.0.0 some text after"; // Compliant
    const string  noOnlyAnIp3 = "Some text before 0.0.0.0 some text after"; // Compliant
```

Maybe you should also add tests on IP+port strings like "192.168.0.1:8080".

Regards

**Comment by guwirth** [comment]
@sgallou can't say much to this check, please have a look to the source code:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/cxx-checks/src/main/java/org/sonar/cxx/checks/HardcodedIpCheck.java

Java plugin version:
https://github.com/SonarSource/sonar-java/blob/master/java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java

Python plugin has also such check:
https://github.com/SonarSource/sonar-python/blob/master/python-checks/src/main/java/org/sonar/python/checks/HardcodedIPCheck.java

Maybe you can compare it and make a proposal for an improvement.

**Comment by Bertk** [comment]
A solution might be to ignore the IP-address "0.0.0.0" or a list of IP addresses but there are some generic checks which allows to add customized checks on line or file scope.
Please check whether you want to resolve this issue with other checkers (“**_File RegEx rule_**” or “**_Line RegEx rule_**”) or you suppress the issue adding a comment at the end of the line ```// NOSONAR```.
Enhancing the checker for rule “**_IP addresses should not be hardcoded_**” with some configurable items is of course another option.

**Comment by guwirth** [comment]
@sgallou I close this. In case you have a better version send us a pull request.
